xtable-core/src/main/java/org/apache/xtable/delta/DeltaSchemaExtractor.java

```diff
@@ -61,6 +61,11 @@ public class DeltaSchemaExtractor {
   private static final String DELTA_COLUMN_MAPPING_ID = "delta.columnMapping.id";
   private static final String COMMENT = "comment";
   private static final DeltaSchemaExtractor INSTANCE = new DeltaSchemaExtractor();
+  // Timestamps in Delta are microsecond precision by default
```

Comment:
Does XTable need to handle nanoseconds precision?

Reply:
Microseconds is the maximum precision supported by spark DataType and AVRO schema, iceberg v3 is the only one that supports nano second precision, we can add it if there's interest in community.